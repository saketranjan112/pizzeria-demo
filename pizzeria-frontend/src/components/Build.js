import React, { useState, useEffect } from 'react';
import axios from 'axios';

export default function Build() {
    const [toppingData, setToppingData] = useState([]);
    const [selectedToppings, setSelectedToppings] = useState([]);
    const [totalCost, setTotalCost] = useState(0);
    const [user, setUser] = useState(JSON.parse(localStorage.getItem('user'))?.result);

    const handleCheck = (e, price, name) => {
        console.log(price)
        console.log(e.target.checked)
        if(e.target.checked){
            setTotalCost(totalCost + price);
            setSelectedToppings([...selectedToppings, name]);
        }else{
            setTotalCost(totalCost - price);
            const newArray = selectedToppings.filter((topping) => topping !== name);
            setSelectedToppings(newArray);
        }
        console.log(selectedToppings)
    }

    useEffect(() => {
        axios.get('http://localhost:4000/toppings')
        .then((response) => setToppingData(response.data))
        .catch((error) => console.log(error));

        setUser(JSON.parse(localStorage.getItem('user'))?.result);
    },[])

    const handleClick = () => {
        console.log(totalCost)
        console.log(selectedToppings)
        console.log(user.name, user.cartId)
        const pizzaName = "Naruto's Recipe";
        const cartId = user.cartId;
        axios.post('http://localhost:4000/cart/addCustomToCart',{ cartId: cartId, name: pizzaName, toppings: selectedToppings, price: totalCost})
        .then((response) => console.log(response))
        .catch((error) => console.log(error));
    }
    
  return (
    <div className="container">
        <p className="text-secondary my-2 text-center">Pizzeria now gives you options to build your own pizza. Customize your pizza by choosing ingredients from the list given below.</p>
        <table className="table table-sm table-bordered table-striped align-middle mx-auto rounded" style={{"width": "70%"}}>
            <tbody>
                {toppingData.map((topping) => {
                    return(
                        <tr key={topping.id}>
                            <td>
                                <img src={topping.image} className="card-img-top" alt="..." style={{"height": 55, "width": 55}}/>
                            </td>
                            <td>
                                <span>{topping.tname}</span>&nbsp;&nbsp; 
                                <span>(<b><sup>&#x20B9;</sup>{topping.price}.00</b>)</span>
                            </td>
                            <td>
                                <div className="form-check d-flex justify-content-center">
                                    <input className="form-check-input" type="checkbox" defaultChecked={false} onChange={(e) => handleCheck(e, topping.price, topping.tname)} />&nbsp;&nbsp;
                                    <label className="form-check-label text-warning" htmlFor="flexCheckDefault">
                                        Add
                                    </label>
                                </div>
                            </td>
                        </tr>
                    )
                })}
                <tr>
                    <td colSpan="3" rowSpan="2">
                        <p className="fs-3 text-secondary text-start mt-2">
                            Total Cost : <span className="text-success"><sup>&#x20B9;</sup>{totalCost}.00</span>
                        </p>
                        <div className="d-grid">

                            { user ? selectedToppings.length < 3 && <p style={{"fontSize": 12}} className="text-danger">Please select a minimum of 3 toppings to build your pizza.</p> 
                            : <p style={{"fontSize": 12}} className="text-danger">Please Signin to build your pizza.</p>}
                            <button className="btn btn-outline-dark bg-warning" disabled={selectedToppings.length < 3} onClick={handleClick}>
                                <div>Build Pizza</div> 
                                <div style={{"fontSize": 12}}>& Add to cart</div>
                            </button>
                        </div>
                    </td>
                </tr>
                
            </tbody>
        </table>
        
    </div>
  )
}
