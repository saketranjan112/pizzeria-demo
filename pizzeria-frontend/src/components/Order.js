import axios from 'axios';
import React, { useEffect, useState } from 'react';
import * as api from '../api/index';

export default function Order({updateCart}) {
    const [pizzaData, setPizzaData] = useState([]);
    const [user, setUser] = useState(JSON.parse(localStorage.getItem('user'))?.result);

    useEffect(() => {
        axios.get('http://localhost:4000/pizzas')
        .then((response) => setPizzaData(response.data))
        .catch((error) => console.log(error));

        setUser(JSON.parse(localStorage.getItem('user'))?.result);
    }, [])

    const addToCartHandler = async (pizza, cartId) => {
        
        await api.addToCart(pizza, cartId);


        updateCart((prevState) => {
            console.log('updateCart called');
            return prevState + 1;
        });
    }
    
  return (
      <div className="container">
        <div className="row justify-content-center">
            {pizzaData.map((pizza) => {
                return(
                    <div className="card col-5 mt-3 pt-2 mx-2" key={pizza.id}>
                        <div className="row">
                            <div className="col-3">
                                <h5 className="card-title">{pizza.name}</h5>
                                <div className={pizza.type === 'veg' ? 'border border-2 border-success mx-auto' : 'border border-2 border-danger mx-auto'} style={{"height": 20, "width": 20}}>
                                    <div className={pizza.type === 'veg' ? 'bg-success rounded-circle mx-auto mt-1' : 'bg-danger rounded-circle mx-auto mt-1'} style={{"height": 10, "width": 10}}></div>
                                </div>
                                <p className="card-text mt-4"><b><sup>&#x20B9;</sup>{pizza.price}.00</b></p>
                            </div>
                            <div className="col-5 text-start" style={{"fontSize": "12px"}}>
                                <p>{pizza.description}</p>
                                <p><b>Ingredients:</b> {pizza.ingredients.map((ingredient, index) => <span key={index}>{ingredient},&nbsp;</span>)}</p>
                                <p><b>Toppings:</b> {pizza.topping.map((topping, index) => <span key={index}>{topping},&nbsp;</span>)}</p>
                            </div>
                            <div className="col-4">
                                <img src={pizza.image} className="card-img-top" alt="..." style={{"height": 140, "width": 140}}/>
                                <br/><br/>
                                <button href="#" className="btn btn-warning text-white" disabled={!user} onClick={() => addToCartHandler(pizza, user.cartId)}>Add to Cart</button>
                                {!user && <p className='text-danger' style={{"fontSize": "8px"}}>Sign in for adding items to cart.</p>}
                            </div>
                        </div>
                    </div>
                )
            })}
        </div>
      </div>
  )
}