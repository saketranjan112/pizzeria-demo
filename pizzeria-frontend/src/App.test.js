import  { configure, render, mount, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Order from './components/Order';
import Build from './components/Build';
import ReactTestRenderer from 'react-test-renderer';


configure({ adapter: new Adapter()});

describe('<Foo />', () => {
    it('renders order component', () => {
      const tree = ReactTestRenderer.create(<Order/>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it('renders build component', () => {
        const tree = ReactTestRenderer.create(<Build/>).toJSON();
        expect(tree).toMatchSnapshot();
      });
  });