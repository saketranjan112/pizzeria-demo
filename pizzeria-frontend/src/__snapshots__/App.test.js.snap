// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Foo /> renders build component 1`] = `
<div
  className="container"
>
  <p
    className="text-secondary my-2 text-center"
  >
    Pizzeria now gives you options to build your own pizza. Customize your pizza by choosing ingredients from the list given below.
  </p>
  <table
    className="table table-sm table-bordered table-striped align-middle mx-auto rounded"
    style={
      Object {
        "width": "70%",
      }
    }
  >
    <tbody>
      <tr>
        <td
          colSpan="3"
          rowSpan="2"
        >
          <p
            className="fs-3 text-secondary text-start mt-2"
          >
            Total Cost : 
            <span
              className="text-success"
            >
              <sup>
                â‚¹
              </sup>
              0
              .00
            </span>
          </p>
          <div
            className="d-grid"
          >
            <p
              className="text-danger"
              style={
                Object {
                  "fontSize": 12,
                }
              }
            >
              Please Signin to build your pizza.
            </p>
            <button
              className="btn btn-outline-dark bg-warning"
              disabled={true}
              onClick={[Function]}
            >
              <div>
                Build Pizza
              </div>
              <div
                style={
                  Object {
                    "fontSize": 12,
                  }
                }
              >
                & Add to cart
              </div>
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<Foo /> renders order component 1`] = `
<div
  className="container"
>
  <div
    className="row justify-content-center"
  />
</div>
`;
